### PR TITLE
Build: Simplify tsconfig.json files

### PR DESCRIPTION
**Pull request by sirreal**
## What?

Simplify tsconfig.json files. [TypeScript 5.5 introduced a `configDir` variable](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#the-configdir-template-variable-for-configuration-files) that can be used for templating tsconfig.json files. This can vastly reduce repitition in tsconfig.json files.

- Add standardized configuration to base tsconfig.json file.
- Remove redundant configuration from package tsconfig.json files.
- Remove redundant file globs (e.g. `src` instead of `src/**/*`).

## Why?

This removes a lot of repetition from tsconfig.json files.

## Testing Instructions
CI passes. TSC should continue to work as expected locally.

### CODE REVIEW

File: bin/validate-tsconfig.mjs [hunk 1]
```diff
@@ -29,14 +29,33 @@ for ( const packageName of packagesWithTypes ) {
...
-		)
-	);
+	let packageJson;
+	try {
+		packageJson = JSON.parse(
+			readFileSync( `packages/${ packageName }/package.json`, 'utf8' )
+		);
+	} catch ( e ) {
+		console.error(
+			`Error parsing package.json for package ${ packageName }`
```
Comment (manzoorwanijk) [inline bin/validate-tsconfig.mjs:39]: Thank you for handling errors here.

File: packages/a11y/tsconfig.json [hunk 2]
```diff
@@ -1,10 +1,5 @@
 {
 	"$schema": "https://json.schemastore.org/tsconfig.json",
```
Comment (manzoorwanijk) [inline packages/a11y/tsconfig.json:2]: While we are here, why do we need this "$schema" in ever `tsconfig.json` file? Don't the IDEs and TS automatically infer that?

**Review by manzoorwanijk — APPROVED** [review]
This is a nice clean up. I have been thinking about this for a while. Thank you for beating me.

File: packages/a11y/tsconfig.json (hunk 2 above)
Comment (manzoorwanijk) [reply packages/a11y/tsconfig.json:2]: I see that it was added in #59861

Comment (sirreal) [reply packages/a11y/tsconfig.json:2]: Some might infer it based on the filename, but including it ensures that supporting environments will use the appropriate schema. There's little downside to including it and the benefits are significant.

**Comment by manzoorwanijk** [comment]
We need to fix `rootDir` for `test/e2e/tsconfig.json` because there is no `src` directory. I will create a PR to add a couple more improvements there.

**Comment by manzoorwanijk** [comment]
I have created #68346 following this one.

**Pull request merged**